variables.tf

```diff
@@ -41,26 +52,31 @@ variable "database_users" {
 # mongodbatlas_advanced_cluster
 
 variable "cluster_name" {
-  type    = string
-  default = "Cluster0"
+  description = "Name of the cluster as it appears in Atlas."
+  type        = string
+  default     = "Cluster0"
 }
 
 variable "provider_name" {
-  type    = string
-  default = "AWS"
+  description = "Cloud service provider on which the servers are provisioned. Possible values are AWS, GCP, AZURE or TENANT."
+  type        = string
+  default     = "AWS"
 }
 
 variable "backing_provider_name" {
-  type    = string
-  default = "AWS"
+  description = "Cloud service provider on which you provision the host for a multi-tenant cluster."
+  type        = string
+  default     = "AWS"
 }
 
 variable "region_name" {
-  type    = string
-  default = "US_EAST_1"
+  description = "Physical location of your MongoDB cluster (more information on https://www.mongodb.com/docs/atlas/cloud-providers-regions/)."
+  type        = string
+  default     = "US_EAST_1"
 }
 
 variable "electable_specs" {
+  description = "Hardware specifications for electable nodes in the region."
```

Comment:
Should we mention that if provider is TENANT instance size should be M5 or lower? ([reference documentation](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Clusters/operation/createCluster)).

Reply:
I tried to follow AWS, Oracle, etc. convention where they only use a brief sentence to explain the variable. I would mention it in the global readme